### ISSUE TITLE
Optional vibe.d support

### ISSUE THREAD

**Issue by MartinNowak** [daehlith/statsd4d#3]
How about optionally supporting async vibe.d sockets?
It's fairly simple, you just need to use version blocks.

``` d
version (Have_vibe_d)
{
    import vibe.core.net;
    //
}
else
{
    import std.socket;
    //
}
```

http://vibed.org/blog/posts/writing-native-db-drivers
http://vibed.org/api/vibe.core.net/UDPConnection

**Comment by daehlith** [comment]
Funny that, as I was writing this I briefly considered making the underlying socket class a template parameter. Eventually I decided this wouldn't be worthwhile since telemetry data should only use UDP sockets so I skipped working in that direction.

I wasn't aware that vibe.d had its own socket implementation. Let me think about this a bit more, even though D offers the version keyword then I'm not necessarily a big fan of it. That being said it's certainly a very pragmatic way of supporting different kinds of sockets that might not share the same interface - though in that case the underlying transmission code might need changes as well.
